src/main/webapp/app/core/auth/state-storage.service.ts

```diff
@@ -8,7 +8,7 @@ export class StateStorageService {
     /**
      * Get the previous state of the current session.
      */
-    getPreviousState(): string | null {
+    getPreviousState(): string {
```

Comment:
Wait! What will `this.sessionStorage.retrieve` return if there is no value for `previousState` defined?

Reply:
I guess it's fine, because this.sessionStorage.retrieve returns `any`

Reply:
but you are actually right, that the method implementation can return `null` so we should probably stay with the previous definition here and add a comment that null is ok here